### ISSUE TITLE
Improve Examples of Triangulation_2 and Triangulation_3

### ISSUE THREAD

**Issue by afabri** [CGAL/cgal#275]
Many examples only insert a sequence of points and print the number of faces. 
We should come up with examples that cover more of the API

**Comment by GilesBathgate** [comment]
Yes please, especially how you would "recover the whole set the triangle belongs to 
by walking from triangle to triangle, never crossing a constrained edge." as discussed here:
http://cgal-discuss.949826.n4.nabble.com/polygon-triangulation-tt3356157.html#a3474956

I did manage to get something working by looking at Nef_Polyhedron_3.h class Triangulation_handler2

My traversal code is here: https://github.com/GilesBathgate/RapCAD/commit/eb6347aa3a78e7e8010353c630fc7a63ddf51158

**Comment by sgiraudot** [comment]
@MoniqueTeillaud, as maintainer of `Triangulation_3`, what do you think about this issue?

My opinion is that we should first discuss which features we want to see in the examples and then see how to fit them nicely in the documentation.

**Comment by MoniqueTeillaud** [comment]
Sorry for my late answer. I am not well organized for these emails through github, there are too many CGAL mailing lists to follow. I don't understand why we don't discuss such topics on cgal-develop and/or during CGAL developers meetings. 
Andreas' comment is very general. How could I disagree with the general idea of improving examples... 
However, we should be aware that not evey single function of a CGAL class can possibly be illustrated by an example. Which ones do we consider as more important, or maybe more difficult to use, so that it would require an example?

**Comment by sgiraudot** [comment]
@afabri what is your opinion on that?

**Comment by afabri** [comment]
I will at last add an example where  one sees the API of Vertex and Face.

**Comment by MoniqueTeillaud** [comment]
I am not convinced. The API of vertex and cell is not the hardest part of the API of T3, IMHO. 
And even the basic example of T3 shows some/most of the API of vertex and cell: 
  Vertex_handle v = c->vertex( (li+1)&3 );
  Cell_handle nc = c->neighbor(li);
  assert( nc->has_vertex( v, nli ) );
What else...?
